engines/batch-ensembler/ensembler/components/experimentation.py

```diff
@@ -0,0 +1,57 @@
+from abc import abstractmethod
+from typing import Optional, Any, Union
+import numpy
+import pandas
+import mlflow
+
+PREDICTION_COLUMN_PREFIX = '__predictions__'
+
+
+class Ensembler(mlflow.pyfunc.PythonModel):
+
+    def load_context(self, context):
+        self.initialize(context.artifacts)
+
+    @abstractmethod
+    def initialize(self, artifacts: dict):
+        """
+        Implementation of Ensembler can specify initialization step which
+        will be called one time during model initialization.
+        :param artifacts: dictionary of artifacts passed to log_component method
+        """
+        pass
+
+    def predict(self, context, model_input: pandas.DataFrame) -> \
+            Union[numpy.ndarray, pandas.Series, pandas.DataFrame]:
+        prediction_columns = {col: col[len(PREDICTION_COLUMN_PREFIX):]
+                              for col in model_input.columns if col.startswith(PREDICTION_COLUMN_PREFIX)}
+        return model_input \
+            .rename(columns=prediction_columns) \
+            .apply(lambda row:
+                   self.ensemble(
+                       features=row.drop(prediction_columns.values()),
+                       predictions=row[prediction_columns.values()],
+                       treatment_config=None
```

Comment:
This is for a future work. Whether it will be in `model_input` or not is out of scope for now, but at some point this batch ensembler should be integrated with an experiment engine and `treatment_config` should be the part of this `ensemble` method.